packages/hooks/src/use_future.rs

```diff
@@ -8,10 +8,34 @@ use dioxus_signals::*;
 use dioxus_signals::{Readable, Writable};
 use std::future::Future;
 
-/// A hook that allows you to spawn a future
-///
+/// A hook that allows you to spawn a future.
+/// This future will **not** run on the server
 /// The future is spawned on the next call to `flush_sync` which means that it will not run on the server.
 /// To run a future on the server, you should use `spawn` directly.
+/// `use_future` assumes your future will never complete - **it won't return a value**.
```

Comment:
Use future can be used with both futures that stop or run forever. It doesn't require the future to run forever

Reply:
Most of the text for these snippets was paraphrased from the examples folder, this one specifically from future.rs at  https://github.com/DioxusLabs/dioxus/blob/main/examples/eval.rs. I will change the comments inside that file too, so that they can both be correct.